app/jobs/api_key_inactive_warning_job.rb

```diff
@@ -5,13 +5,18 @@ class ApiKeyInactiveWarningJob < ApplicationJob
 
   def perform
     activity_period = 11.months.ago
-    active_api_keys = GoogleAnalyticsQueries.get_active_api_keys_from_ga(activity_days)
+    active_api_keys = GoogleAnalyticsQueries.get_active_api_keys_from_ga(activity_period)
+    User.where(api_key: active_api_keys).update_all(sent_warning_email: false)
     api_keys_older_than_activity_period = User.where(user_type: 'api')
                                                 .where.not(email: nil)
                                                 .where("created_at <= :date", date: activity_period)
                                                 .pluck(:api_key)
     inactive_api_keys = api_keys_older_than_activity_period - active_api_keys
-    inactive_api_users = User.where(api_key: inactive_api_keys)
-    inactive_api_users.map { |u| NotifyMailer.api_key_inactive_warning(u).deliver_later }
+  
+    users_to_warn = User.where(api_key: inactive_api_keys, sent_warning_email: false)
+    users_to_warn.map { |u|
```

Comment:
Minor: given that you are not keeping the result of the map, it would be better to use `#each` instead. Also it's considered better style to use do/end for multiline blocks:

```ruby
users_to_warn.each do |u|
  NotifyMailer.api_key ...
  u.update ...
end
```

Reply:
If I'm not mistaken, `NotifyMailer.api_key_inactive_warning` raises an exception if the request to notify is not successful. Should this be handled here?